compiler/pom.xml

```diff
@@ -198,6 +202,23 @@
                   <pattern>com.google.auto.common</pattern>
                   <shadedPattern>dagger.shaded.auto.common</shadedPattern>
                 </relocation>
+                <relocation>
+                  <pattern>com.squareup.javapoet</pattern>
+                  <shadedPattern>dagger.shaded.javapoet</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>com.google.googlejavaformat</pattern>
+                  <shadedPattern>dagger.shaded.googlejavaformat</shadedPattern>
+                </relocation>
+                <!-- Brought in by google-java-format 0.1-alpha -->
```

Comment:
Ugh, but this could get pretty messy if we need to shade all transitive deps. Maybe the right thing to do is to shade all the deps and use `ClassName.get("com.google.common.util", "ListenableFuture")` like @JakeWharton suggested.